subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstyleExtension.java

```diff
@@ -113,7 +116,18 @@ public File getConfigDir() {
      */
     @Incubating
     public void setConfigDir(File configDir) {
-        this.configDir = configDir;
+        this.configDir.set(configDir);
+    }
+
+    /**
+     * Gets the configuration directory.
+     *
+     * @return The configuration directory
+     * @since 4.7
+     */
+    @Incubating
+    public DirectoryProperty getConfigDirectory() {
```

Comment:
We need to add this to `org.gradle.api.plugins.quality.CheckstyleExtension.xml`

Reply:
done